### ISSUE TITLE
BxBlock without name fails to rebuild

### ISSUE THREAD

**Issue by aveXcaesar** [icarus-consulting/BriX#13]
### Expected Behavior
`BxBlock` without name can be printed with `RebuildMedia`

### Actual Behavior
`new BxBlock(new BxProp("test", "x")).Print(new RebuildMedia());` throws an exception:
```
System.ArgumentException: ''' is an invalid expanded name.'
```

### Mention any other details that might be useful
The following is working:
`new BxBlock("missingName", new BxProp("test", "x")).Print(new RebuildMedia());`
results in:
```xml
<missingName bx-type="block">
  <test bx-type="prop">x</test>
</missingName>
```

### The log given by the failure.

**Comment by aveXcaesar** [comment]
Discussion on Slack:
__hwi  13:46__
Ja die Exception ist falsch. Ein BxBlock muss immer einen Namen haben, außer er wird in ein BxBlockArray gesteckt

**caesar  13:48**
Wäre es dann gut den ctor ohne Namen zu entfernen?

**hwi  13:48**
Das gleiche mit JsonMedia ergibt: System.InvalidOperationException : You are trying to make a block without a name into a property or an object. Blocks without a name can only be put into arrays.  Die Exception bräuchte man bei den anderen medias auch
**13:49**
Nein denn für das BxBlockArray benötigt man BxBlocks ohne namen

**caesar  13:50**
Dann benutzt man BxBlock ggf. falsch und merkt es erst, wenn man ein Media benutzt
**13:50**
Sollte man nicht das BxBlockArray umbauen? (edited) 

**hwi  13:52**
Spätestens im Unittest sollte es ja auffallen... Ja könnte man mal diskutieren wie man die benutzung intuitiver gestaltet.
new BxBlock("root", new BxBlock(new BxProp("test", "x"))).Print(new RebuildMedia());  Mit einem Root Knoten herum ist die meldung übrigens: System.InvalidOperationException : You are trying to make a block without a name into a property or an object. Blocks without a name can only be put into arrays.

**Comment by HerickJ-22** [comment]
I don't think the expected behavior should be `BxBlock without name can be printed with RebuildMedia`. The exception message should be fixed. 
Without a name the question is, what name the node should have, and if a random name is selected by the BxBlock that would be FailSave and not FailFast

**Comment by aveXcaesar** [comment]
My approach is to remove the ctor without a name. Is there a problem with `BxBlockArray` or any others?

**Comment by HerickJ-22** [comment]
Yes, because if you put a BxBlock without a name into a BxBlockArray, the BxBlockArray throws a Exception, if the name of the block is different to the name of the items in the BxBlockArray.
This would not only be a breaking change, but also remove the usability that the Block without a name can be put into any array no matter how the entries name in the array is

**Comment by icaOBU** [comment]
I think till we improved the usability of BxBlockArrays, we shouldnt´t remove the ctor and change the exception message. 
Maybe we can introdue a seperate object for arrays? I'm not shure if it´s a good way. If you have other ideas, please feel free to make suggestions :)
That the exception occures not until the brix is printed to a media works like designed. We aggregate the data only at this time.

**Comment by aveXcaesar** [comment]
Issue updated:
The question is how to use this library:
In my opinion, you should not be able to create a BxBlock that cannot also be printed.
But it is possible to create a BxBlock as root element, which leads to the given exception.
The question arises whether the knowledge should be presupposed, how the Library is to be used, or whether a more intuitive use is wanted.

**Comment by Meerownymous** [comment]
It works as it was designed. The empty ctor is designed to be a shorthand for usage in arrays, as was stated above.
Normally, that should result in the appropriate error message that has been quoted by @HerickJ-22.

I do not see a need for another object, the usage of brix is simple and the error message should be sufficient to solve the problem.

**Comment by HerickJ-22** [comment]
@Meerownymous The errormessage has been fixed with #17 
This Issue was open for discussion, because it is not clear for the user that a BxBlock without a name can not be build as root

**Comment by Meerownymous** [comment]
@HerickJ-22 Thanks for clarification. As stated, I think a meaningful error msg is sufficient to prevent misusage in this case, like it is in other places of the library too.
